File: Lab2/Person.m
//
//  Person.m
//  Lab2
//
//  Created by Derrick Park on 2022-05-10.
//

#import "Person.h"
// - : instance method
// + : static method
@interface Person()

@property (nonatomic) NSString *username;

@end

@implementation Person {
  // by convention, private "instance variable" prefixed with _name
//  NSString *_name;
//  NSInteger _age+ (void) staticMethod;
}

- (instancetype)initWithName: (NSString *) name andAge: (NSInteger) age
{
  self = [super init];
  if (self) {
    _name = name;
    _age = age;
    _username = @"";
  }
  return self;
}

- (void)printPersonInfo {
  NSLog(@"Name: %@, Age: %ld", _name, _age);
}

- (void)dealloc {
  NSLog(@"Gone!");
}

@end


// .h : header file - 'public'
// .m : implementation file - code 

File: Lab2/Person.swift
//
//  Person.swift
//  Lab2
//
//  Created by Derrick Park on 2022-05-10.
//

import Foundation

class Person {
  // property(instance variable + getter/setter)
  private(set) var name: String
  private(set) var age: Int
  
  init(name: String, age: Int) {
    self.name = name
    self.age = age
  }
  
  func printPersonInfo() {
    print("Name: \(name), Age: \(age)")
  }
}

